### ISSUE TITLE
Regression on SetRequired

### ISSUE THREAD

**Issue by raphaelboukara** [sindresorhus/type-fest#1020]
### Bug description

Let's take the following code has an example:
```ts
import type { PackageJson, SetRequired } from 'type-fest';

type PackageJsonVersion = PackageJson['version'];
type PackageJsonVersionRequired = SetRequired<PackageJson, 'version'>['version'];
```

### received
With `type-fest@4.30.2` I get:
```
type PackageJsonVersion = string | undefined

type PackageJsonVersionRequired = ((string | number | boolean | JsonObject | JsonValue[] | readonly JsonValue[]) & (string | number | boolean | JsonObject | JsonValue[] | readonly JsonValue[] | undefined)) | null
```

### expected
With `type-fest@4.30.1` I get:
```
type PackageJsonVersion = string | undefined

type PackageJsonVersionRequired = string
```

### Repro

https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBDAnmApnA3nACgQwMYDWOA5igFIDOEAdgDRwDKKMASigI4CuwUKAJnAC+cAGZQIIOAHIkqALQiUFGFIDcAWABQW2WlyES5KtQBqKKBWA04AXmz4ipSjQDaUgG7nLNKQF0Nmrr2Bk7GZhZW1GxcPPy2jMzR3Lx8ADz6jkY09B5ekVIAfG6eET7+QA

**Comment by som-sm** [comment]
This is related to index signatures, should be fixed with [this PR](https://github.com/sindresorhus/type-fest/pull/1014).

**Comment by raphaelboukara** [comment]
Thanks a lot
Fixed in [4.31.0](https://github.com/sindresorhus/type-fest/commit/de218af1b2d9806c1ccaefed8e3a64f0921dd244).
